pkg/status/status.go

```diff
@@ -85,10 +87,45 @@ func GetStatus() (map[string]interface{}, error) {
 	}
 
 	stats["systemProbeStats"] = getSystemProbeStats()
+	stats["apmStats"] = GetAPMStatus()
 
 	return stats, nil
 }
 
+// GetAPMStatus returns a set of key/value pairs summarizing the status of the trace-agent.
+// A key "outcome" is always returned and it may be either "error" or "success".
+func GetAPMStatus() map[string]interface{} {
+	port := 8126
+	if p, ok := os.LookupEnv("DD_APM_RECEIVER_PORT"); ok {
+		if v, err := strconv.Atoi(p); err == nil {
+			port = v
+		}
+	}
+	if config.Datadog.IsSet("apm_config.receiver_port") {
+		port = config.Datadog.GetInt("apm_config.receiver_port")
+	}
```

Comment:
This  should be handle by the config package. At startup we should overwrite `apm_config.receiver_port` with `DD_APM_RECEIVER_PORT` if needed. and here only do `port = config.Datadog.GetInt("apm_config.receiver_port")`

Reply:
You're right, this isn't ideal. Unfortunately we don't have a choice because all of the env. vars used by APM are defined outside the shared config package [here](https://github.com/DataDog/datadog-agent/blob/1a3ab11e293927bb4a5112a52623989347b9c36b/pkg/trace/config/env.go#L21-L22). This is so because of the way we load the legacy configuration for Agent 5. Arguably that code could be removed and then we could apply your suggestions but I think that should be a job for another PR. I'll leave a TODO note.

Reply:
We should at least pull the default from the configuration.

And I'm missing why we cant move `DD_APM_RECEIVER_PORT` to the config package for example ? All we do is binding the env key [here](https://github.com/DataDog/datadog-agent/blob/1a3ab11e293927bb4a5112a52623989347b9c36b/pkg/trace/config/env.go#L52), or I'm missing something ?

Reply:
Ah yes, you're right. Damn. Sorry about that. I was confused for a bit because of some other confusing problems with env. vars. Fixed.

Reply:
That code is never called by the core agent. As you've noticed yourself, this all happens in the core agent binary. We need to move all environment variable bindings from APM into the main `config` package.

I'm not going to bore you with endless details, but this was not possible because of Agent 5. Now it is, and we should do it. I will do it as a follow up PR.